node/network/gossip-support/src/lib.rs

```diff
@@ -107,26 +140,213 @@ impl GossipSupport {
 						err = ?e,
 						"Failed to receive a message from Overseer, exiting",
 					);
-					return
+					return self
 				},
 			};
 			match message {
-				FromOverseer::Communication { .. } => {},
+				FromOverseer::Communication {
+					msg: GossipSupportMessage::NetworkBridgeUpdateV1(ev),
+				} => self.handle_connect_disconnect(ev),
 				FromOverseer::Signal(OverseerSignal::ActiveLeaves(ActiveLeavesUpdate {
 					activated,
 					..
 				})) => {
 					tracing::trace!(target: LOG_TARGET, "active leaves signal");
 
 					let leaves = activated.into_iter().map(|a| a.hash);
-					if let Err(e) = state.handle_active_leaves(&mut ctx, &keystore, leaves).await {
+					if let Err(e) = self.handle_active_leaves(&mut ctx, leaves).await {
 						tracing::debug!(target: LOG_TARGET, error = ?e);
 					}
 				},
 				FromOverseer::Signal(OverseerSignal::BlockFinalized(_hash, _number)) => {},
-				FromOverseer::Signal(OverseerSignal::Conclude) => return,
+				FromOverseer::Signal(OverseerSignal::Conclude) => return self,
+			}
+		}
+	}
+
+	/// 1. Determine if the current session index has changed.
+	/// 2. If it has, determine relevant validators
+	///    and issue a connection request.
+	async fn handle_active_leaves<Context>(
+		&mut self,
+		ctx: &mut Context,
+		leaves: impl Iterator<Item = Hash>,
+	) -> Result<(), util::Error>
+	where
+		Context: SubsystemContext<Message = GossipSupportMessage>,
+		Context: overseer::SubsystemContext<Message = GossipSupportMessage>,
+	{
+		// Use active leaves update as tick to see whether we need to report on connectivity.
+		self.check_connectivity();
```

Comment:
why not use a stream of every 10mins in `run`? we don't pass any params from active leaves signal

Reply:
Fair. :-)  I am just used to that natural way of polling we have due to `ActiveLeavesUpdate`.